merkletree/pad.go

```diff
@@ -21,12 +21,12 @@ type PAD struct {
 	snapshots    map[uint64]*SignedTreeRoot
 	loadedEpochs []uint64 // slice of epochs in snapshots
 	latestSTR    *SignedTreeRoot
-	policies     Policies // the current policies in place
+	policies     *Policies // the current policies in place
```

Comment:
`pad.policies` is in use, so the same mutation comment applies here.

I think you should revert everything in this patch that isn't strictly about removing the interface, sorry.